test/Tester/CancellationTests/CancellationTokenPassingTests.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.TestingHost;
+using Orleans.Threading;
+using UnitTests.GrainInterfaces;
+using Xunit;
+using UnitTests.Tester;
+
+namespace UnitTests.MembershipTests
+{
+    public class CancellationTokenPassingTests : HostedTestClusterPerTest
+    {
+        private readonly double[] cancellationDelaysInMS = { 0, 10, 50, 500 };
+
+        public override TestingSiloHost CreateSiloHost()
+        {
+            return new TestingSiloHost(new TestingSiloOptions
+            {
+                StartFreshOrleans = false,
```

Comment:
We should never set `StartFreshOrleans` this as false in our tests... This property will probably be removed after a new refactoring. The point is this will not do what you intend, and especially since you used `HostedTestClusterPerTest`